File: tests/parsers/samples/code/kt/complex.kt
package samples

import kotlin.math.abs
import kotlin.math.sqrt

data class ComplexMetrics(val scale: Double = 1.0) {
    fun distance(a: Int, b: Int): Double = abs(a - b) * scale

    fun rootMeanSquare(values: List<Double>): Double {
        if (values.isEmpty()) return 0.0
        val total = values.sumOf { it * it }
        return sqrt(total / values.size)
    }

    companion object {
        fun buildDefault() = ComplexMetrics(0.86)
    }
}

File: tests/parsers/samples/code/scala/complex.scala
package samples

object ComplexMetrics {
  def distance(a: Int, b: Int, scale: Double = 1.0): Double = math.abs(a - b) * scale

  def rootMeanSquare(values: List[Double]): Double = {
    if (values.isEmpty) 0.0
    else math.sqrt(values.map(v => v * v).sum / values.length)
  }
}
